testgrid/cmd/updater/main.go

```diff
@@ -298,26 +328,38 @@ type Row struct {
 	Result   state.Row_Result
 	Metrics  map[string]float64
 	Metadata map[string]string
+	Message  string
+	Icon     string
 }
 
-func (br Column) Overall() state.Row_Result {
+func (br Column) Overall() Row {
+	r := Row{
+		Metadata: map[string]string{"Tests name": "Overall"},
+	}
 	switch {
 	case br.Finished > 0:
-		// Completed
+		// Completed, did we pass?
 		if br.Passed {
-			return state.Row_PASS
+			r.Result = state.Row_PASS // Yep
+		} else {
+			r.Result = state.Row_FAIL
+		}
+		r.Metrics = map[string]float64{
+			elapsedKey: float64(br.Finished - br.Started),
 		}
-		return state.Row_FAIL
 	case time.Now().Add(-24*time.Hour).Unix() > br.Started:
 		// Timed out
-		return state.Row_FAIL
+		r.Result = state.Row_FAIL
+		r.Message = "Never finished"
```

Comment:
Maybe "Test didn't finish in 24 hours" or similar?

Reply:
Done